src/Runners/PHPUnit/ResultPrinter.php

```diff
@@ -122,6 +126,9 @@ public function __construct(LogInterpreter $results, OutputInterface $output, Op
         $this->output  = $output;
         $this->options = $options;
 
+        $this->printsTeamcity = $this->options->teamcity();
+        $this->needsTeamcity = $this->printsTeamcity || $this->options->logTeamcity();
```

Comment:
Same here: output and logging are two different things and shouldn't be coupled

Reply:
I have moved this check into a method in the Options: https://github.com/paratestphp/paratest/pull/597/files#diff-e7eb5918ab64b6a45aac1cb6586bd195c5a96fd600882a7be2478afe58edbe5cR1108